rotkehlchen/db/history_events.py

```diff
@@ -238,32 +239,39 @@ def reset_eth_staking_data(
             self.db.delete_dynamic_caches(write_cursor=write_cursor, key_parts=key_parts)
 
     @staticmethod
-    def reset_evm_events_for_redecode(
+    def reset_events_for_redecode(
             write_cursor: 'DBCursor',
-            location: EVM_EVMLIKE_LOCATIONS_TYPE,
+            location: BLOCKCHAIN_LOCATIONS_TYPE,
     ) -> None:
-        """Reset EVM events and transaction decode status for the given location.
-
-        Deletes all non-customized EVM events and marks their transactions
-        as not decoded to enable re-processing.
+        """Reset the given location's events, etc. for re-decoding.
+        Handles different cases depending on the location:
+        * Bitcoin - simply deletes all non-customized bitcoin events.
+        * EVM and EVM-like - deletes non-customized events that also have a corresponding
+          transaction in the evm_transactions table.
+        * EVM - removes the EVMTX_DECODED evm_tx_mappings to enable re-processing.
         """
         customized_events_num = write_cursor.execute(
             'SELECT COUNT(*) FROM history_events_mappings WHERE name=? AND value=?',
             (HISTORY_MAPPING_KEY_STATE, HISTORY_MAPPING_STATE_CUSTOMIZED),
         ).fetchone()[0]
+        evm_join = (
```

Comment:
rename this variable. This is not for evm only. It needs a different name. And also it's not a join if it's bitcoin, it's a where. So you need to choose an appropriate name.